var/spack/repos/builtin/packages/py-fenics-dolfinx/package.py

```diff
@@ -49,32 +52,38 @@ class PyFenicsDolfinx(PythonPackage):
     depends_on("fenics-dolfinx@0.1.0", when="@0.1.0")
 
     depends_on("fenics-basix@main", type=("build", "link"), when="@main")
+    depends_on("fenics-basix@0.6.0:0.6", type=("build", "link"), when="@0.6.0:0.6")
     depends_on("fenics-basix@0.5.1:0.5", type=("build", "link"), when="@0.5.0:0.5")
     depends_on("fenics-basix@0.4.2", type=("build", "link"), when="@0.4.1")
     depends_on("fenics-basix@0.3.0", type=("build", "link"), when="@0.3.0")
     depends_on("fenics-basix@0.2.0", type=("build", "link"), when="@0.2.0")
     depends_on("fenics-basix@0.1.0", type=("build", "link"), when="@0.1.0")
 
     depends_on("py-fenics-ffcx@main", type="run", when="@main")
+    depends_on("py-fenics-ffcx@0.6", type="run", when="@0.6")
     depends_on("py-fenics-ffcx@0.5.0.post0", type="run", when="@0.5.0:0.5")
     depends_on("py-fenics-ffcx@0.4.2", type="run", when="@0.4.1")
     depends_on("py-fenics-ffcx@0.3.0", type="run", when="@0.3.0")
     depends_on("py-fenics-ffcx@0.2.0", type="run", when="@0.2.0")
     depends_on("py-fenics-ffcx@0.1.0", type="run", when="@0.1.0")
 
     depends_on("py-fenics-ufl@main", type="run", when="@main")
-    depends_on("py-fenics-ufl@2022.2.0", type="run", when="@0.5.0:0.5")
-    depends_on("py-fenics-ufl@2022.1.0", type="run", when="@0.4.1")
-    depends_on("py-fenics-ufl@2021.1.0", type="run", when="@0.1:0.3.99")
+    depends_on("py-fenics-ufl@2023.1", type="run", when="@0.6")
```

Comment:
Shouldn't these all be type=build/run?

Reply:
Won't the import tests (`spack install --test=run py-fenics-dolfinx`) require these to be in `PYTHONPATH` in order to pass?

Reply:
yes, I don't think any tests are currently implemented, but this is something I intend to do soon. As far as I'm aware, testing isn't doing what we want currently, so I have held back (https://github.com/spack/spack/issues/29447). I could probably write some C++ tests, so yes ffcx at build time makes sense in this case, if I understand spack install --test=root correctly.

Reply:
Yes, only the import tests are implemented by default.

Reply:
I will keep this in mind for a forthcoming PR. Can v0.6.0 be merged, or shall I amend this now in anticipation of future tests?

Reply:
The import tests already require these to be present at build time, so I don't see why we shouldn't make them build/run deps now.